src/fsharp/ConstraintSolver.fs

```diff
@@ -155,21 +155,21 @@ type ContextInfo =
     /// The type equation comes from a sequence expression.
     | SequenceExpression of TType
 
-exception ConstraintSolverTupleDiffLengths of DisplayEnv * TType list * TType list * range  * range 
-exception ConstraintSolverInfiniteTypes of ContextInfo * DisplayEnv * TType * TType * range * range
-exception ConstraintSolverTypesNotInEqualityRelation of DisplayEnv * TType * TType * range * range * ContextInfo
-exception ConstraintSolverTypesNotInSubsumptionRelation of DisplayEnv * TType * TType * range  * range 
-exception ConstraintSolverMissingConstraint of DisplayEnv * Tast.Typar * Tast.TyparConstraint * range  * range 
-exception ConstraintSolverError of string * range * range
-exception ConstraintSolverRelatedInformation of string option * range * exn 
-
-exception ErrorFromApplyingDefault of TcGlobals * DisplayEnv * Tast.Typar * TType * exn * range
-exception ErrorFromAddingTypeEquation of TcGlobals * DisplayEnv * TType * TType * exn * range
-exception ErrorsFromAddingSubsumptionConstraint of TcGlobals * DisplayEnv * TType * TType * exn * ContextInfo * range
-exception ErrorFromAddingConstraint of  DisplayEnv * exn * range
-exception PossibleOverload of DisplayEnv * string * exn * range
-exception UnresolvedOverloading of DisplayEnv * exn list * string * range
-exception UnresolvedConversionOperator of DisplayEnv * TType * TType * range
+exception ConstraintSolverTupleDiffLengths              of displayEnv: DisplayEnv * TType list * TType list * range  * range
```

Comment:
Nit: We don't normally have these sorts of whitespace additions. @dsyme what are your thoughts here?

Reply:
I noticed that but let it slide, it's ok

Reply:
Do you want me to clear the spacing in #6596, the overload related exceptions are changing.

Reply:
Could we clarify if, for example, such alignement is proscribed/prescribed for specific or any situations:

https://github.com/Microsoft/visualfsharp/blob/master/CONTRIBUTING.md#coding-guidelines

Do the maintainers want to set a bar somewhere else than "code should have reasonable format, good effort on readability, and fit in the context"?

How does that get addressed in the guidelines & our CI/review workflow in the future?

I agree that code alignment makes editing slightly more involved in few cases, this is the main drawback I see, but the readability pros generally outweights it (subjective, for me).

I also agree the maintainers have the command on that when it comes to review/integrate contributions.

To look further around us, have you noticed jetbrains has been adding support for advanced settings for a bit: https://blog.jetbrains.com/dotnet/2017/11/28/aligning-code-columns-resharper-rider/ ?

It indicates that the demand for such features exist.

NB: This is not a rant 😄, let me know how to fix the code in the other PR.

Reply:
I personally am not a fan of column alignment (and an F#-specific variant, arrow aligment in match expressions, is not recommended as per the [official style guide](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting))

I don't think we have a hard and fast rule, but uniformity is what's preferred here above all else. That's why lots of whitespace normalization PRs were recently brought in.

Reply:
Yes we've been removing column alignment. There are some places we still use it though, e.g. tast.fs ValFlags, where it is extremely useful

Mostly we've been removing it to add space for doc comments, e.g.
```
let foo1 xxx yyy zzz = align
let foo2 x   y   z = align
```
becomes
```
/// comment
let foo1 xxx yyy zzz = align

/// comment
let foo2 x y z = align
```
Not much point in keeping the alignment when comments are added